src/views/Dashboard/GeneticAnalyst/Account/index.vue

```diff
@@ -695,36 +697,47 @@ export default {
     async onSubmitFile() {
       this._touchForms("document")
 
-      const { title, issuer, month, year, description, supporting_document } = this.document
+      const { title, issuer, month, year, description, file, supportingDocument } = this.document
 
-      if (!title || !issuer || !month || !year || !supporting_document) return this.errorDoc = true
+      if (!title || !issuer || !month || !year) return this.errorDoc = true
```

Comment:
Remove `return` and please update to `if (!title || !issuer || !month || !year) this.errorDoc = true`

Reply:
if remove the `return` code after it will be executed, WDYT @bukandicki

Reply:
then you need to declare it after your `this.errorDoc` like

```javascript
if (!title || !issuer || !month || !year) {
  this.errorDoc = true
  return
}
```

please don't return setter, use return as a value or to prevent something, this is a bad way, so you need to update, even the sonarcloud say it as code smells

Reply:
updated @bukandicki , thanks for the advice